spec/features/consumer/shopping/embedded_shopfronts_spec.rb

```diff
@@ -25,6 +25,7 @@
       Spree::Config[:embedded_shopfronts_whitelist] = 'localhost'
 
       page.driver.browser.js_errors = false
+      allow_any_instance_of(ApplicationController).to receive(:embedded_shopfront_referer).and_return('localhost')
```

Comment:
It's generally a bad practice to stub the class under test, because then the test does not execute the code that you just added. Have you tried stubbing the request instead?

Reply:
I've stubbed the request instead and improved the test.